neurodamus/stimulus_manager.py

```diff
@@ -733,8 +733,13 @@ def __init__(self, target, stim_info: dict, cell_manager):
                 if not sc.exists():
                     continue
 
-                # create single electrode voltage clamp at location
-                seclamp = Nd.h.SEClamp(tpoint_list.x[sec_id], sec=sc.sec)
+                # If conductanceSource not available, insert standard SEClamp
+                if hasattr(Nd.h, "conductanceSource"):
+                    seclamp = Nd.h.conductanceSource(tpoint_list.x[sec_id], sec=sc.sec)
+                else:
+                    # create single electrode voltage clamp at location
+                    seclamp = Nd.h.SEClamp(tpoint_list.x[sec_id], sec=sc.sec)
+
```

Comment:
The question raised by @joseph-tharayil in https://github.com/BlueBrain/neurodamus-models/pull/6#issuecomment-1994629078 . 

>  a user might want to simulate a "real" SEClamp (i.e., use SEClamp to simulate a physical electrode rather than to account or missing synapses) or even to use it in both ways in the same simulation. In that case the new mod file should not be used. Is there a way to allow the user to specify whether or not they want it?

I am wondering, shall we add a new key in SEClamp to switch between h.conductanceSource and h.SEClamp ? This key will be added in the config file and the SONATA spec for [seclamp](https://sonata-extension.readthedocs.io/en/latest/sonata_simulation.html#seclamp-voltage-clamp)

Reply:
That seems like a good idea, but having taken a stab at it I think implementing it is a bit beyond my understanding of how neurodamus processes simulation configs. Would someone have the bandwidth this week to help out with that?

Reply:
Hi @joseph-tharayil , sorry for my late answer. I can take care of adding the new key in SONATA config and in neurodamus if needed. 
But before that, I am wondering how different are the simulation results between the current `SEClamp` and the new `conductanceSource`. Which mechanim will be the default one in the future for the "seclamp" stimulus?

Invite @seirios and @andrisecker here for the discussion. Thanks!

Reply:
At least as far as I tested it, the results are identical. The only difference is how the currents are reported. I think the new one should be the default, since my intuition is that users will be more likely to use SEClamp the way we do (to model a missing input) than to model a physical electrode (in which case IClamp would be more appropriate)

Reply:
Actually, it's now occurring to me that we will run into the same problems with IClamp. Right now, it is included in the LFP calculation, but that should not be the case if it is modeling a physical electrode. Maybe the best option is to include a `use_in_lfp` key in the config file for all current/conductance sources. Then, we can use the new mod file in all cases

Reply:
Just wanted to follow up on this, since we're almost ready to publish the BlueRecording paper: 

One comment that we got from other scientists was that it might be confusing if the `seclamp` stimulus does not use `h.seclamp`. I think it would make sense if the `conductanceSource` was the default for the `shot_noise` and `ornstein_uhlenbeck` stimuli only, with a key in the config file to switch back to `h.seclamp`. 

Similarly, for current sources, I think the default behavior should be for the current to be included in the LFP calculation, with an optional flag to exclude it.

Does this sound reasonable?

Reply:
Hi @joseph-tharayil,
If I understand correctly, for the `class ConductanceSource` in `stimli.py`, we will use `h.conductanceSource` mechenism by default, and a flag from the simulation config file to switch to `h.SEClamp`.  
How about the `class CurrerntSource` ? Is there a change on the currect `h.IClamp` ?

Reply:
@WeinaJi yes, that's correct regarding `class ConductanceSource`. For `class CurrentSource`, we don't need to make any change to `h.IClamp`. Currently, the `IClamp` current is added to `i_membrane` when the LFP is calculated. This should remain the default behavior, but there should be a flag in the stimulation config to prevent it from being added in the LFP calculation. @jorblancoa would that be something that is easy to implement?

Reply:
Hi @joseph-tharayil , after talking to @jorblancoa, I think we need 2 new parameters in the simulation config. 
1. For conductance stimulus type, in the tables of [shot_noise, absolute_shot_noise and relative_shot_noise](https://github.com/BlueBrain/sonata-extension/blob/7208597a9b9eb4423feabbc47c355f85770af64f/source/sonata_simulation.rst?plain=1#L346) and [ornstein_uhlenbeck and relative_ornstein_uhlenbeck ](https://github.com/BlueBrain/sonata-extension/blob/7208597a9b9eb4423feabbc47c355f85770af64f/source/sonata_simulation.rst?plain=1#L373), we need a new string parameter which is either `ConductanceSource` or `SEClamp`. I am not sure how to name it, `conductance_type`? Can you think of a better name?
2. As for the `IClamp` current from LFP, we need a new boolean parameter in the [report table](https://github.com/BlueBrain/sonata-extension/blob/7208597a9b9eb4423feabbc47c355f85770af64f/source/sonata_simulation.rst?plain=1#L402) to exclude `IClamp` from the `lfp` report.
 
Because we are not sure about the names of those two new parameters in the SPEC, is it possible that you create a PR in https://github.com/BlueBrain/sonata-extension to add them?  Then we will take care of implementing it in the simulation toolchain.